### ISSUE TITLE
Perspective guide cursor

### ISSUE THREAD

**Issue by Shnatsel** [mypaint/mypaint#152]
Sketchbook Pro added a perspective guide mode in addition to straight lines, ellipses, etc. Here's a demo: https://www.youtube.com/watch?v=r96G15n08Y4

It would be nice to have something similar in MyPaint.

**Comment by gwojcik** [comment]
I created tool for drawing lines in single point perspective. It is created using gui.linemode.
It allow drawing perspective lines from VP, horizontal and vertical lines. VP is set by clicking with ctrl key pressed. With shift it draw line form VP.

I would like to work on this tool and add more features unless there are other plans about perspective mode in MyPaint. 
My changes are in branch [perspectiveMode] (https://github.com/gwojcik/mypaint/commits/perspectiveMode).

Things that I think will be good additions to perspective mode:
- Possibility to determine direction in 3D space that is represented by VP. This will require setting center of projection, and circle of view. This will help in maintaining relations between multiple sets of VPs.
- Remembering multiple sets of VPs.

**Comment by iirelu** [comment]
![blender](https://cloud.githubusercontent.com/assets/3432543/5603273/55fb47be-9375-11e4-94fd-aed38f0a4a43.png)

It looks like a cool start! There definitely needs to be some visual feedback, though, but that's a little harder. It took me a while to figure out how it actually *worked* (first click with the tool specifies the vanishing point, and then drawn lines snap either towards it or snap horizontally/vertically.

A lot of work needs to be done (visual feedback, continuous snapped line drawing instead of rendering a line from start to end) but it's the kind of stuff where we don't exactly have the underlying framework to implement it.

Thanks for all your effort, but making this as good as it deserves to be would be a bunch harder, and in the end this is still a lower priority than many other things.

**Comment by Shnatsel** [comment]
PaintStorm Studio allows changing brush parameters (most importantly size) based on perspective guides set up by the user. It's really convenient to get perspective to affect brush size, as shown here: (warning - loud music) https://youtu.be/oXUReniLyqw?t=50s

**Comment by Shnatsel** [comment]
Turns out Krita implements a very similar thing already: https://docs.krita.org/Painting_With_Assistants

**Comment by achadwick** [comment]
@gwojcik 
I'm really glad to see that work has continues on this feature. Are you still interested in working on merging this? Now would be a very good time, and I have time to review :smile: 

(We now make more of a thing of assigning issues people, so should I assign this one to you for 1.3.0?)

**Comment by gwojcik** [comment]
@achadwick 
Yes, I'm still interested in continuing this.
There are still some things that need improvements, like point editing, optimized line drawing and user interface.
I have also some changes that are not pushed to GitHub (possibility to determine direction represented by vanishing pint).

> Now would be a very good time, and I have time to review

Should I prepare changes for merge request?
Final merging of this changes will require plugin system?

> (We now make more of a thing of assigning issues people, so should I assign this one to you for 1.3.0?)

Yes.

**Comment by achadwick** [comment]
@gwojcik 
That's great news. Yay for new features! :tulip: 

> Final merging of this changes will require plugin system?

Not necessarily, after all we already have features that might be turned into plugins one day. And an assemblage of Overlay, Mode etc. classes constituting a tool is almost a standard unit in MyPaint these days.

I'll assign this one to you shortly: adding you as an external collaborator with read access first so I can do that.

**Comment by AesaraB** [comment]
I don't think this feature is too far an offshoot from adding things like French curves and rulers
